### ISSUE TITLE
Error while using GPU based execution

### ISSUE THREAD

**Issue by prabhujana** [tensorflow/federated#816]
I am trying to experiment with remote executor runtime with the example provided on this link.
https://github.com/tensorflow/federated/blob/master/tensorflow_federated/python/examples/remote_executor_example.py

If I using CPU based tensorflow, then everything works fine. However, for GPU based tensorflow 
the follow error occurs and aborts execution:

2020-03-29 16:27:22.904103: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2020-03-29 16:27:22.904807: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 978 MB memory) -> physical GPU (device: 0, name: GRID V100DX-32C, pci bus id: 0000:02:00.0, compute capability: 7.0)
2020-03-29 16:27:22.995000: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Internal: No unary variant device copy function found for direction: 1 and Variant type_index: tensorflow::data::(anonymous namespace)::DatasetVariantWrapper
	 [[{{node partitionedcall_args_0/_2}}]]

How do I solve this ? Have anyone faced similar issues ?

**Comment by cramertj** [comment]
Sorry you hit this! There's an internal bug tracking this issue as well, and it's currently blocked on some work from the mainstream Tensorflow team. I'll try to give an update here when there's more to say, and thanks for your patience!

**Comment by tzapzoor** [comment]
Hi! I also ran into this bug while trying to run tensorflow federated with a GPU base remote executor. Is there any way around it?

Some remote executor logs:
```
2020-05-03 12:21:38.334182: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1241] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10770 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:00:04.0, compute capability: 3.7)
2020-05-03 12:21:38.682825: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Internal: No unary variant device copy function found for direction: 1 and Variant type_index: tensorflow::data::(anonymous namespace)::DatasetVariantWrapper
	 [[{{node partitionedcall_args_0/_2}}]]
2020-05-03 12:21:38.760455: W tensorflow/core/common_runtime/base_collective_executor.cc:217] BaseCollectiveExecutor::StartAbort Internal: No unary variant device copy function found for direction: 1 and Variant type_index: tensorflow::data::(anonymous namespace)::DatasetVariantWrapper
	 [[{{node partitionedcall_args_0/_2}}]]
```

**Comment by cramertj** [comment]
Sadly still no update internally. :(

**Comment by jkr26** [comment]
Hi all! For the version of this we have hit internally, we have been able to work around the underlying TF issue with [this commit](https://github.com/tensorflow/federated/commit/15fb10008921447690d1ea5a42829b9f9a7e832b). Can those who have experienced this issue externally confirm whether or not this resolves their problem? At this point, this would necessitate building TFF from master--but we are looking to have a pip package out "soon" (I am on rotation next week and promised to make a release). I will ping this bug back when we have a pip package out as well.

**Comment by jkr26** [comment]
This should be fixed as of [TFF 0.14.0](https://github.com/tensorflow/federated/releases/tag/0.14.0), just released. Would it be possible to confirm?

**Comment by jkr26** [comment]
We believe this issue to be resolved as of TFF 0.14.0; closing now.
